core/i18n/intl.ts

```diff
@@ -0,0 +1,12 @@
+import areIntlLocalesSupported from 'intl-locales-supported'
+
+export const importIntlPolyfill = async () => {
+  const IntlPolyfill = await import('intl')
+  global.Intl = IntlPolyfill.default
+}
+
+export const checkForIntlPolyfill = async (storeView) => {
+  if (!global.Intl || !areIntlLocalesSupported(storeView.i18n.defaultLocale)) {
```

Comment:
shouldn't we need to check window also :thinking: ? `!(window || global).Intl` and same in line 5

Reply:
Good point, there are browsers which don't support `Intl` –  like mobile Safari: https://caniuse.com/#search=Intl